src/foraging_gui/Foraging.py

```diff
@@ -1990,6 +1992,290 @@ def _about(self):
+        save_me['PhotometryFolder']=self.PhotometryFolder
+        save_me['MetadataFolder']=self.MetadataFolder
+        
+        # get the open ephys recording information
+        save_me['open_ephys'] = self.open_ephys
+
+        # Check to see if any frames were dropped
+        # Placing this here so that changes will be tracked and saved.
+        self._check_drop_frames(save_tag= 1)# If this is expensive we should skip.
+        save_me['drop_frames_tag']=self.drop_frames_tag
+        save_me['trigger_length']=self.trigger_length
+        save_me['drop_frames_warning_text']=self.drop_frames_warning_text
+        save_me['frame_num']=self.frame_num
+
+        return save_me
+
+    def _SaveInitialSettings(self):
+        """Save the current settings"""
+        # Create a new folder, if necessary
+        if self.CreateNewFolder==1:
+            self._GetSaveFolder()
+            self.CreateNewFolder=0
+        save_me = self._ConstructSettingsObject()
+        self.CurrentSettings = copy.copy(save_me)
+
+        current_time = datetime.now()
+        formatted_datetime = current_time.strftime("%Y-%m-%d_%H-%M-%S")
+        save_me['_whoami'] = 'settings'
+        save_me['_whenami'] = formatted_datetime
+        
+        # Get the file name to save to.
+        this_settings_json = f'settings_init_{self.ID.text()}_{formatted_datetime}'
+        full_output_file = os.path.join(self.CheckpointFolder, this_settings_json + '.json')
+        with open(full_output_file, "w") as outfile:
+           json.dump(save_me, outfile, indent=4, cls=NumpyEncoder)
+        # Store settings to check for differential later.
+        logging.info(f"Saved settings to {full_output_file}")
+
+    def _SaveSettingsChange(self):
+        save_me = self._ConstructSettingsObject()
+        # Check if the settings have changed
+        if hasattr(self, 'CurrentSettings'):
+            changed_obj={}
+            for key in save_me.keys():
+                if key not in self.CurrentSettings.keys():
+                    changed_obj[key] = save_me[key]
+                elif save_me[key] != self.CurrentSettings[key]:
+                    changed_obj[key] = save_me[key]
+            current_time = datetime.now()
+            formatted_datetime = current_time.strftime("%Y-%m-%d_%H-%M-%S")
+            changed_obj['_whoami'] = 'settings'
+            changed_obj['_whenami'] = formatted_datetime
+            this_settings_json = f'settings_{self.GeneratedTrials.B_CurrentTrialN}_{self.ID.text()}_{formatted_datetime}'
+            full_output_file = os.path.join(self.CheckpointFolder, this_settings_json + '.json')
+            with open(full_output_file, "w") as outfile:
+                json.dump(changed_obj, outfile, indent=4, cls=NumpyEncoder)
+        else:
+            error('CurrentSettings not found. Cannot save settings change. This should not happen, means code was called in wrong order')
+        self.CurrentSettings = copy.copy(save_me)
```

Comment:
why do you need the copy here?

Reply:
I was having issues where pointed to settings would change but the changes were not tracked when I checked the differential. Copy seemed to resolve.